src/browser/squirrel-update.coffee

```diff
@@ -231,19 +129,24 @@ exports.handleStartupEvent = (app, squirrelCommand) ->
   switch squirrelCommand
     when '--squirrel-install'
       createShortcuts ->
-        installContextMenu ->
+        WinRegistry.installContextMenu ->
           addCommandsToPath ->
             app.quit()
       true
     when '--squirrel-updated'
       updateShortcuts ->
-        installContextMenu ->
+        contextMenuWasEnabled = atom.config.get('core.showAtomInShellContextMenu')
```

Comment:
Ok, got that. You know what, maybe we don't necessarily need the `atom.config` to detect if user had disabled that, at least as long as accessing it is not easy.
During update, so in browser process, we already access the Windows Registry to read/write keys. 
We could just define a way to detect - by reading the registry - whether the context menu is enabled or not. A new method would be added to `WinRegistry` to this purpose. Then, during update, `SquirrellUpdate` would acces such method, instead of `atom.config`.
What do you think about that?